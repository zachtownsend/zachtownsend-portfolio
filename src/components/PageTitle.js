import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const StyledPageTitle = styled.h1`
  color: ${(props) => props.theme.white};
  font-size: 24px;
  line-height: 44px;
  text-align: left;

  .site-title {
    display: none;
    color: ${(props) => props.theme.lightGray};

    + .page-title {
      margin-left: 10px;
    }
  }

  .page-title::before {
    content: "/";
    color: ${props => props.theme.primary};
    display: inline-block;
    margin-right: 10px;
  }

  @media (min-width: ${props => props.theme.breakpoints.tablet}px) {
    text-align: center;

    .site-title {
      display: inline-block;
    }
  }
`;

function PageTitle(props) {
  if(!props.pageTitle && !props.siteTitle) return null;

  return (
    <header className={props.className} data-testid="page-title">
      <StyledPageTitle>
        {props.siteTitle && <span className="site-title">{props.siteTitle}</span>}
        {props.pageTitle && <span className="page-title">{props.pageTitle}</span>}
      </StyledPageTitle>
    </header>
  )
}

PageTitle.propTypes = {
  siteTitle: PropTypes.string,
  pageTitle: PropTypes.string,
};

export default PageTitle

