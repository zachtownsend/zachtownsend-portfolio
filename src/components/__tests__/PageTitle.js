import React from 'react';
import { render } from "react-testing-library"
import PageTitle from '../PageTitle';

describe("PageTitle", () => {
  it("Shows the site title and the page title separated by a slash", () => {
    const Test = <PageTitle pageTitle="Projects" siteTitle="Zach Townsend" />;

    const { getByText } = render(Test);

    const parent = getByText(/\//);
    const { children } = parent;

    expect(children.length).toBe(2);
    expect(children[0].textContent).toBe('Zach Townsend');
    expect(children[1].textContent).toBe('Projects');
  });

  it('Only shows the site title if page title omitted', () => {
    const Test = <PageTitle siteTitle="Zach Townsend" />;
    const { container } = render(Test);
    const h1 = container.querySelector('h1');

    expect(h1).toBeDefined();
    expect(h1.children.length).toBe(1);
    expect(h1.firstChild.textContent).toBe('Zach Townsend');
    expect(h1.firstChild.className).toBe('site-title');
  });

  it('Only shows the page title if site title omitted', () => {
    const Test = <PageTitle pageTitle="Projects" />;
    const { container } = render(Test);
    const h1 = container.querySelector('h1');

    expect(h1).toBeDefined();
    expect(h1.children.length).toBe(1);
    expect(h1.children[0].textContent).toBe('Projects');
    expect(h1.children[0].className).toBe('page-title');
  });

  it('has h1 as the first parent', () => {
    const Test = <PageTitle pageTitle="Projects" siteTitle="Zach Townsend" />;
    const { container } = render(Test);
    const h1 = container.querySelector('h1');
    expect(h1).toBeDefined();
  });

  it('displays null if neither props are entered', () => {
    const Test = <PageTitle />;
    const { queryByTestId } = render(Test);
    expect(queryByTestId('page-title')).toBeNull();
  });
});
