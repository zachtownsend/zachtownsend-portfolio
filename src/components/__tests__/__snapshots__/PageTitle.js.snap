// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageTitle Shows the site title and the page title separated by a slash 1`] = `<header />`;
